lib/event-stream.js

```diff
@@ -138,84 +145,94 @@ EventStream.prototype.doLongPoll = function() {
 	};
 
 	this._longPollRetries += 1;
-	this._client.get(url, options, this._client.defaultResponseHandler(function(err, data) {
-
-		if (err) {
-			setTimeout(function() {
-				self.getLongPollInfo();
-			}, RETRY_DELAY);
-			return;
-		}
+	return this._client.wrapWithDefaultHandler(this._client.get)(url, options)
+		.then(data => {
 
-		if (data.message === 'reconnect') {
-			self.getLongPollInfo();
-			return;
-		}
+			if (data.message === 'reconnect') {
+				this.getLongPollInfo();
+				return;
+			}
 
-		// We don't expect any messages other than reconnect and new_change, so if
-		// we get one just retry the long poll
-		if (data.message !== 'new_change') {
-			self.doLongPoll();
-			return;
-		}
+			// We don't expect any messages other than reconnect and new_change, so if
+			// we get one just retry the long poll
+			if (data.message !== 'new_change') {
+				this.doLongPoll();
+				return;
+			}
 
-		self.fetchEvents();
-	}));
+			this.fetchEvents();
+		})
+		.catch(() => {
+			setTimeout(() => this.getLongPollInfo(), this._options.retryDelay);
+		});
 };
 
 /**
  * Fetch the latest group of events and push them into the stream
- * @returns {void}
+ * @returns {Promise} Promise for testing purposes
  * @private
  */
 EventStream.prototype.fetchEvents = function() {
 
-	var self = this,
-		eventParams = {
-			stream_position: this._streamPosition
-		};
-
-	this._client.events.get(eventParams, function(err, events) {
-
-		if (err) {
-			self.emit('error', err);
-			setTimeout(function() {
-				self.getLongPollInfo();
-			}, RETRY_DELAY);
-			return;
-		}
-
-		// If the response wasn't what we expected, re-poll
-		if (!events.entries || !events.next_stream_position) {
-			self.doLongPoll();
-			return;
-		}
-
-		self._streamPosition = events.next_stream_position;
-
-		// De-duplicate the fetched events, since the API often returns
-		// the same events at multiple subsequent stream positions
-		var newEvents = events.entries.filter(function(event) {
-			return !self._dedupHash[event.event_id];
-		});
-
-		// If there aren't any non-duplicate events, go back to polling
-		if (!newEvents.length) {
-			self.doLongPoll();
-			return;
-		}
-
-		// Push new events into the stream
-		newEvents.forEach(function(event) {
-			self._dedupHash[event.event_id] = true;
-			self.push(event);
-		});
+	var eventParams = {
+		stream_position: this._streamPosition,
+		limit: 500
```

Comment:
I don't think so; for performance we should always take the maximum.  We do the same in the Admin Events stream as well.